import "goby/common/protobuf/option_extensions.proto";
import "goby/common/protobuf/app_base_config.proto";

package goby.moos.protobuf;

message MOOSGatewayConfig
{
  optional AppBaseConfig base = 1;
  
  optional string moos_server_host = 2 [default = "localhost"];
  optional uint32 moos_server_port = 3 [default = 9000];
  optional uint32 moos_comm_tick = 4 [default = 5];
  repeated string moos_subscribe_filter = 5 [(goby.field).description="Substring filter for all MOOS subscriptions; 'NAV_' matches 'NAV_X', 'NAV_Y', etc.; '' matches everything."];
  repeated string goby_subscribe_filter = 6 [(goby.field).description="Substring filter for all Goby (ZeroMQ) subscriptions; 'NAV_' matches 'NAV_X', 'NAV_Y', etc.; '' matches everything."];  
  
}
